packages/shorebird_tests/test/shorebird_tests.dart

```diff
@@ -0,0 +1,204 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:archive/archive_io.dart';
+import 'package:path/path.dart' as path;
+
+import 'package:meta/meta.dart';
+import 'package:test/test.dart';
+
+/// This will be the path to the flutter binary housed in this flutter repository.
+///
+/// Which since we are running the tests from this inner package , we need to go up two directories
+/// in order to find the flutter binary in the bin folder.
+File get _flutterBinaryFile => File(
+      path.join(
+        Directory.current.path,
+        '..',
+        '..',
+        'bin',
+        'flutter${Platform.isWindows ? '.bat' : ''}',
+      ),
+    );
+
+/// Runs a flutter command using the correct binary ([_flutterBinaryFile]) with the given arguments.
+Future<ProcessResult> _runFlutterCommand(
+  List<String> arguments, {
+  required Directory workingDirectory,
+  Map<String, String>? environment,
+}) async {
+  return Process.run(
+    _flutterBinaryFile.absolute.path,
+    arguments,
+    workingDirectory: workingDirectory.path,
+    environment: {
+      'FLUTTER_STORAGE_BASE_URL': 'https://download.shorebird.dev',
+      if (environment != null) ...environment,
+    },
+  );
+}
+
+Future<void> _createFlutterProject(Directory projectDirectory) async {
+  final result = await _runFlutterCommand(
+    ['create', '.'],
```

Comment:
minor but you can use `--empty` here to create a simplified sample
```suggestion
    ['create', '--empty', '.'],
```